packages/headless/src/features/tab-set/tab-set-slice.ts

```diff
@@ -18,15 +19,23 @@ export const tabSetReducer = createReducer(
       })
       .addCase(updateActiveTab, (state, action) => {
         const id = action.payload;
-        const hasId = id in state;
-
-        if (!hasId) {
-          return;
-        }
-
-        Object.keys(state).forEach((tabId) => {
-          state[tabId].isActive = tabId === id;
-        });
+        activateTabIfIdExists(state, id);
+      })
+      .addCase(restoreSearchParameters, (state, action) => {
+        const id = action.payload.tab || '';
+        activateTabIfIdExists(state, id);
       });
   }
 );
+
+function activateTabIfIdExists(state: TabSetState, id: string) {
+  const hasId = id in state;
+
+  if (!hasId) {
```

Comment:
Similar to the other PR for tab, I'm not a huge fan of silent return when something is "obviously" wrong with the state of headless VS what the reducers are processing.